File: Samples/GridLoaders/DNN/VB/GridLoader/Utilities.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls

Imports DotNetNuke.Common
Imports DotNetNuke.Entities.Portals
Imports DotNetNuke.Services.Exceptions

Namespace EntitySpaces.Modules.GridLoader

    ''' <summary>
    ''' Summary description for Utilities
    ''' </summary>
    Public Class Utilities
        ' private ctor all functions are static
        Private Sub New()
        End Sub

        Public Shared Function ConstructUrl(ByVal ParamArray args As String()) As String
            Dim ps As PortalSettings = PortalController.GetCurrentPortalSettings()
            Return Globals.NavigateURL(ps.ActiveTab.TabID, String.Empty, args)
        End Function

        Public Shared Function ConstructUrl(ByVal tabId As Integer, ByVal ParamArray args As String()) As String
            Return Globals.NavigateURL(tabId, String.Empty, args)
        End Function
    End Class

End Namespace

File: Samples/GridLoaders/DNN/C#/GridLoader/Utilities.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

using DotNetNuke.Common;
using DotNetNuke.Entities.Portals;
using DotNetNuke.Services.Exceptions;

namespace EntitySpaces.Modules.GridLoader
{

	/// <summary>
	/// Summary description for Utilities
	/// </summary>
	public class Utilities
	{
		// private ctor all functions are static
		private Utilities() {}

		public static string ConstructUrl(params string[] args)
		{
			PortalSettings ps = PortalController.GetCurrentPortalSettings();
			return Globals.NavigateURL(ps.ActiveTab.TabID, string.Empty, args);
		}

		public static string ConstructUrl(int tabId, params string[] args)
		{
			return Globals.NavigateURL(tabId, string.Empty, args);
		}
	}

}
